base_multi_image/models/image.py

```diff
@@ -0,0 +1,181 @@
+             "or kanban views.")
+    image_small = fields.Binary(
+        "Small-sized image",
+        compute="_get_image_sizes",
+        help="Small-sized image. It is automatically resized as a 64 x 64 px "
+             "image, with aspect ratio preserved. Use this field anywhere a "
+             "small image is required.")
+    comments = fields.Text(
+        'Comments',
+        translate=True)
+    show_technical = fields.Boolean(
+        compute="_show_technical")
+
+    @api.multi
+    @api.depends('storage', 'path', 'file_db_store', 'url')
+    def _get_image(self):
+        """Get image data from the right storage type."""
+        for s in self:
+            s.image_main = getattr(s, "_get_image_from_%s" % s.storage)()
+
+    @api.multi
+    @api.depends("owner_id", "owner_model")
+    def _show_technical(self):
+        """Know if you need to show the technical fields."""
+        self.show_technical = all(
+            "default_owner_%s" % f not in self.env.context
+            for f in ("id", "model"))
+
+    @api.multi
+    def _get_image_from_db(self):
+        return self.file_db_store
+
+    @api.multi
+    def _get_image_from_file(self):
+        if self.path and os.path.exists(self.path):
+            try:
+                with open(self.path, 'rb') as f:
+                    return base64.b64encode(f.read())
+            except Exception as e:
+                _logger.error("Can not open the image %s, error : %s",
+                              self.path, e, exc_info=True)
+        else:
+            _logger.error("The image %s doesn't exist ", self.path)
+
+        return False
+
+    @api.multi
+    def _get_image_from_url(self):
+        if self.url:
+            try:
+                (filename, header) = urllib.urlretrieve(self.url)
+                with open(filename, 'rb') as f:
+                    return base64.b64encode(f.read())
+            except:
+                _logger.error("URL %s cannot be fetched", self.url,
+                              exc_info=True)
+
+        return False
+
+    @api.multi
+    @api.depends('image_main')
+    def _get_image_sizes(self):
+        for s in self:
+            try:
+                vals = tools.image_get_resized_images(
+                    s.with_context(bin_size=False).image_main)
+            except:
+                vals = {"image_medium": False,
+                        "image_small": False}
+            s.update(vals)
+
+    @api.multi
+    def _check_filestore(self):
+        """check if the filestore is created, and do it otherwise."""
+        for s in self:
+            dir_path = os.path.dirname(s.path)
+            try:
+                if not os.path.exists(dir_path):
+                    os.makedirs(dir_path)
+            except OSError as e:
+                raise exceptions.Warning(
+                    _('The image filestore cannot be created, %s') % e)
+
+    @api.model
+    def _make_pretty(self, name):
```

Comment:
Hehe, yes, this method doesn't make anybody in a photo image to look prettier :stuck_out_tongue_closed_eyes: